sign-action.md

```diff
@@ -0,0 +1,75 @@
+- You have created a registry in Azure Container Registry
+- You have a GitHub repository to store the sample workflow and GitHub Secret
+
+### Add provider credentials to Github Secrets
+
+Add two credentials to authenticate with ACR and AKV as follows, two Github Secrets are required in the whole process:
+
+- `ACR_PASSWORD`: the password to log in to the ACR where your artifact will be released
+- `AZURE_CREDENTIALS`: the credential to AKV where your key pair is stored
+    
+### Create service principal and generate Azure credential
+
+- Execute the following command to create the service principal and generate Azure credentials. 
+
+```
+# login using your own account
+az login
+
+# Create a service principal
+spn=notationtest
+az ad sp create-for-rbac -n $spn --sdk-auth
+```
+
+> [!IMPORTANT]
+> 1. Add the JSON output of the above `az ad sp` command to [Github Secret](https://learn.microsoft.com/en-us/azure/developer/github/connect-from-azure?tabs=azure-portal%2Cwindows#add-the-service-principal-as-a-github-secret) with name `AZURE_CREDENTIALS`.
+>
+> 2. Save the `clientId` from the JSON output into an environment variable (without double quotes) as it will be needed in the next step:
+>```
+>  clientId=<clientId_from_JSON_output_of_last_step>
+>```
+
+### Create GitHub Secret to store credentials 
+
+- Create an encrypted secret to store the Azure Credentials in your own GitHub repository. See [GitHub Docs](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository) for details.
+
+- Add the JSON output of the following `az ad sp` command to the value of GitHub Secret. Naming the GitHub Secret as `AZURE_CREDENTIALS`. 
+
+- Similarly, create another encrypted secret to store the registry credential or password in your own GitHub repository. For example, we use `ACR_PASSWORD` to store the password of the ACR registry in the GitHub Repository secret.
+
+### Grant AKV permission to the service principal
+
+Grant AKV permission to the service principal that we created in the previous step.
+
+```
+# set policy for your AKV
+akv=<your_akv_name>
+az keyvault set-policy --name $akv --spn $clientId --certificate-permissions get --key-permissions sign --secret-permissions get
+```
+
+See [az keyvault set-policy](https://learn.microsoft.com/en-us/cli/azure/keyvault?view=azure-cli-latest#az-keyvault-set-policy) for details.
+
+### Edit the sample workflow
+
+- Create a `workflows` folder under `.github` and create a `workflow.yml` to run and test the CI/CD pipeline. You can copy the [signing template workflow](https://github.com/notation-playground/notation-integration-with-ACR-and-AKV/blob/template/sign-template.yml) to your own `workflow.yml` file. 
```

Comment:
```suggestion
- Create `<your_repo>/.github/workflows/<your_workflow>.yml` to run and test the CI/CD pipeline. You can copy the [signing template workflow](https://github.com/notation-playground/notation-integration-with-ACR-and-AKV/blob/template/sign-template.yml) to your own `<your_workflow>.yml` file. 
```

Reply:
Based on our discussion, we will include the contents of the `signing template workflow` directly in this doc.